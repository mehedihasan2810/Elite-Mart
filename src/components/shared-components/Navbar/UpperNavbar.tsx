import React from "react";
import styles from "@/scss/shared-component-styles/Navbar/UpperNavbar.module.scss";
import Image from "next/image";
import Link from "next/link";
const UpperNavbar = () => {
  return (
    <div className={styles.upperNavbar_container}>
      <div>

        <div>
          <Image data-testid='flag' src="us.svg" width={25} height={25} alt="us flag" />
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="w-6 h-6"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="M8.25 4.5l7.5 7.5-7.5 7.5"
            />
          </svg>
        </div>

        <div>
          <span>USD</span>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="w-6 h-6"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="M8.25 4.5l7.5 7.5-7.5 7.5"
            />
          </svg>
        </div>
      </div>
      <ul role="list">
        <li>
        <Link href="#">Sign In</Link>
        </li>
       <li>
       <Link href="#">Sign Up</Link>
       </li>
       <li>
       <Link href="#">Our Stores</Link>
       </li>
        <li>
        <Link href="#">Help</Link>
        </li>
       
      </ul>
    </div>
  );
};

export default UpperNavbar;
