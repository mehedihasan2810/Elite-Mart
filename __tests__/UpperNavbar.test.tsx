import UpperNavbar from "@/components/shared-components/Navbar/UpperNavbar";
import { render, screen } from "@testing-library/react";

describe("<UpperNavbar/>", () => {
  test("should UpperNavbar component render properly and USD text and flag img should be in the document", () => {
    // ARRANGE
    const { asFragment } = render(<UpperNavbar />);

    // ACT
    const USD = screen.getByText("USD");
    const flag = screen.getByTestId("flag");

    // ASSERT
    expect(USD).toBeInTheDocument();
    expect(flag).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  test("there should be 4 Link", () => {
    // ARRANGE
    const { asFragment } = render(<UpperNavbar />);

    // ACT
    const linkList = screen.getAllByRole("link");

    // ASSERT
    expect(linkList).toHaveLength(4);
    expect(asFragment()).toMatchSnapshot();
  });
});
