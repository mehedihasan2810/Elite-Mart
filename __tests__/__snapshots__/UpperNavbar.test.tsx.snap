// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpperNavbar/> should UpperNavbar component render properly and USD text and flag img should be in the document 1`] = `
<DocumentFragment>
  <div
    class="upperNavbar_container"
  >
    <div>
      <div>
        <img
          alt="us flag"
          data-nimg="1"
          data-testid="flag"
          decoding="async"
          height="25"
          loading="lazy"
          src="us.svg"
          style="color: transparent;"
          width="25"
        />
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div>
        <span>
          USD
        </span>
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
    <ul
      role="list"
    >
      <li>
        <a
          href="/api/auth/signin"
        >
          Sign In
        </a>
      </li>
      <li>
        <a
          href="/api/auth/signout"
        >
          Sign Out
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Our Stores
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Help
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<UpperNavbar/> there should be 4 Link 1`] = `
<DocumentFragment>
  <div
    class="upperNavbar_container"
  >
    <div>
      <div>
        <img
          alt="us flag"
          data-nimg="1"
          data-testid="flag"
          decoding="async"
          height="25"
          loading="lazy"
          src="us.svg"
          style="color: transparent;"
          width="25"
        />
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div>
        <span>
          USD
        </span>
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5l7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
    <ul
      role="list"
    >
      <li>
        <a
          href="/api/auth/signin"
        >
          Sign In
        </a>
      </li>
      <li>
        <a
          href="/api/auth/signout"
        >
          Sign Out
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Our Stores
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Help
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
